doc/tutorial/metaprogramming-arith.md

```diff
@@ -0,0 +1,333 @@
+## Arithmetic as an embedded domain-specific language
+
+Let's parse another classic grammar, the grammar of arithmetic expressions with
+addition, multiplication, integers, and variables.  In the process, we'll learn
+how to:
+
+- Convert identifiers such as `x` into strings within a macro.
+- add the ability to "escape" the macro context from within the macro. This is useful to interpret identifiers with their _original_ meaning (predefined values)
+  instead of their new meaning within a macro (treat as a symbol).
+
+Let's begin with the simplest thing possible. We'll define an AST, and use operators `+` and `*` to denote
+building an arithmetic AST.
+
+
+Here's the AST that we will be parsing:
+
+```lean,ignore
+-- example on parsing arith language via macros
+inductive Arith: Type where
+ | add : Arith -> Arith -> Arith -- e + f
```

Comment:
```suggestion
inductive Arith : Type where
  | add : Arith -> Arith -> Arith -- e + f
```
one more try :)

Reply:
Oh, this time I forgot the Unicode arrows myself

Reply:
The [latest commit]( https://github.com/bollu/lean4/blob/b152e9da2b81952403ecf6ee06640433c5259fe0/doc/tutorial/metaprogramming-arith.md?plain=1#L17-L23) should hopefully be correct  (spacing of ` : `, 2-spaces before `|`, and unicode arrows)